File: 202-happy-number/happy-number.cpp
class Solution {
public:
    bool isHappy(int n) {
      bool mark=true;
      unordered_set<int>st;
      while(n!=1){
        if(st.find(n)!=st.end()) return false;
        st.insert(n);
        long long ans=0;
        while(n!=0){
            ans+=pow(n%10,2);
            n/=10;
        }
        n=ans;
      }
      return true;
    }
};

File: 202-happy-number/happy-number.java
class Solution {
    public boolean isHappy(int n) {
        Set<Integer>set=new HashSet<>();
        for(int i=1;i<100000000;i++){
            int ans=0;
        while(n!=0){
            int rem=n%10;
            ans+=rem*rem;
            n/=10;
        }
        if(ans==1)return true;
        if(set.contains(ans))return false;
        else set.add(ans);
        n=ans;
        }
        return true;
    }
}